crypto/keys/keybase_keyring.go

```diff
@@ -0,0 +1,511 @@
+//CreateMulti for keyring
+func (kb keyringKeybase) CreateMulti(name string, pub tmcrypto.PubKey) (Info, error) {
+	return kb.writeMultisigKey(name, pub), nil
+}
+
+func (kb *keyringKeybase) persistDerivedKey(seed []byte, passwd, name, fullHdPath string) (info Info, err error) {
+	//create master key and derive first key for keyring:
+	masterPriv, ch := hd.ComputeMastersFromSeed(seed)
+	derivedPriv, err := hd.DerivePrivateKeyForPath(masterPriv, ch, fullHdPath)
+	if err != nil {
+		return
+	}
+
+	if passwd != "" {
+		info = kb.writeLocalKey(name, secp256k1.PrivKeySecp256k1(derivedPriv), passwd)
+	} else {
+		pubk := secp256k1.PrivKeySecp256k1(derivedPriv).PubKey()
+		info = kb.writeOfflineKey(name, pubk)
+	}
+	return
+}
+
+// List returns the keys from storage in alphabetical order.
+func (kb keyringKeybase) List() ([]Info, error) {
+	var res []Info
+	keys, err := kb.db.Keys()
+	if err != nil {
+		return nil, err
+	}
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		if strings.HasSuffix(key, infoSuffix) {
+			rawInfo, err := kb.db.Get(key)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(rawInfo.Data) == 0 {
+				return nil, keyerror.NewErrKeyNotFound(key)
+			}
+
+			info, err := readInfo(rawInfo.Data)
+			if err != nil {
+				return nil, err
+			}
+
+			res = append(res, info)
+
+		}
+	}
+	return res, nil
+}
+
+// Get returns the public information about one key.
+func (kb keyringKeybase) Get(name string) (Info, error) {
+	key := infoKey(name)
+	bs, err := kb.db.Get(string(key))
+	if err != nil {
+		return nil, err
+	}
+	if len(bs.Data) == 0 {
+		return nil, keyerror.NewErrKeyNotFound(name)
+	}
+	return readInfo(bs.Data)
+}
+
+func (kb keyringKeybase) GetByAddress(address types.AccAddress) (Info, error) {
+	ik, err := kb.db.Get(string(addrKey(address)))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ik.Data) == 0 {
+		return nil, fmt.Errorf("key with address %s not found", address)
+	}
+	bs, err := kb.db.Get(string(ik.Data))
+	if err != nil {
+		return nil, err
+	}
+	return readInfo(bs.Data)
+}
+
+// Sign signs the msg with the named key.
+// It returns an error if the key doesn't exist or the decryption fails.
+func (kb keyringKeybase) Sign(name, passphrase string, msg []byte) (sig []byte, pub tmcrypto.PubKey, err error) {
+	info, err := kb.Get(name)
+	if err != nil {
+		return
+	}
+
+	var priv tmcrypto.PrivKey
+
+	switch info.(type) {
+	case localInfo:
+		linfo := info.(localInfo)
+		if linfo.PrivKeyArmor == "" {
+			err = fmt.Errorf("private key not available")
+			return
+		}
+
+		priv, err = cryptoAmino.PrivKeyFromBytes([]byte(linfo.PrivKeyArmor))
+		if err != nil {
+			return nil, nil, err
+		}
+
+	case ledgerInfo:
+		linfo := info.(ledgerInfo)
+		priv, err = crypto.NewPrivKeyLedgerSecp256k1Unsafe(linfo.Path)
+		if err != nil {
+			return
+		}
+
+	case offlineInfo, multiInfo:
+		_, err := fmt.Fprintf(os.Stderr, "Message to sign:\n\n%s\n", msg)
+		if err != nil {
+			return nil, nil, err
```

Comment:
that introduces the following error message "result parameter err not in scope at return"

Reply:
Could you use `cmd.PrintErrf()` instead please? You wouldn't need to check the return error

Reply:
> that introduces the following error message "result parameter err not in scope at return"

try with

```go
_, err = fmt.Fprintf(os.Stderr, "Message to sign:\n\n%s\n", msg)
```

instead of 
```go
_, err := fmt.Fprintf(os.Stderr, "Message to sign:\n\n%s\n", msg)
```

note the `:=` vs `=`